### PR TITLE
Updating base library

### PR DESCRIPTION
**Pull request by miker1423**
I re-run SVD2RUST with the latest version (0.16.1) and generated the library, this time it showed no warnings. Also I updated the dependencies.

### CODE REVIEW

File: examples/flash_led_lptmr/memory.x [hunk 1]
```diff
@@ -9,10 +9,3 @@ MEMORY
 /* This is where the call stack will be allocated. */
 /* The stack is of the full descending type. */
 _stack_start = ORIGIN(STACK) + LENGTH(STACK);
-
-/* You can use this symbol to customize the location of the .text section */
-/* If omitted the .text section will be placed right after the .vector_table
-   section */
-/* This is required only on microcontrollers that store some configuration right
-   after the vector table */
-//_stext = ORIGIN(FLASH) + 0x800;
```
Comment (stefanhoelzl) [inline examples/flash_led_lptmr/memory.x]: why did you delete this?

also commit message is not very clear here, you are saying you added something but only deleted something

**Comment by stefanhoelzl** [comment]
you may want to change the `generate` section in the Makefile:
```bash
.PHONY: generate
generate: clean
	svd2rust -i MK64F12.svd
	form -i lib.rs -o src/ && rm lib.rs
	cargo fmt
```

since there are no more warnings, there is no need to create the WARNINGS.md.
Also the azure pipeline will then work correctly.

File: Cargo.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "k64"
-version = "0.1.0"
-authors = ["Stefan Hoelzl <stefan.hoelzl@posteo.de>"]
+version = "0.2.0"
```
Comment (stefanhoelzl) [inline Cargo.toml]: please do not update the version.

I will do this when publishing the crate, to have consistency when the version gets bumped

**Comment by stefanhoelzl** [comment]
instead of deleting the flash example in one commit and then adding it again in another, can you squash both commits into one?

**Comment by stefanhoelzl** [comment]
thanks for your work and contributing!

File: examples/flash_led_lptmr/memory.x (hunk 1 above)
Comment (miker1423) [reply examples/flash_led_lptmr/memory.x]: The commit message is wrong, I wrote it wrong. I removed that section because it caused overlapping of the .text and .rodata sections.

Comment (stefanhoelzl) [reply examples/flash_led_lptmr/memory.x]: ok!

a small hint for your next pull request:
* fork the Project
* create a feature branch for your work
* do your commits on the feature branch
* open the pull request from this feature branch

The advantage is that on this feature branch you can [rewrite your commit Messages or Squash commits together](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) (you do not want to do this on your master branch). 

This way you can organize your commits in a way that they better represent your actual code changes.
